src/TLuaInterpreter.cpp

```diff
@@ -2742,61 +2677,54 @@ int TLuaInterpreter::disableCommandLine(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#replace
 int TLuaInterpreter::replace(lua_State* L)
 {
-    std::string a1 = "";
-    std::string a2 = "";
     int n = lua_gettop(L);
     int s = 1;
     if (!lua_isstring(L, s)) {
         lua_pushstring(L, "replace: wrong argument type");
         lua_error(L);
         return 1;
-    } else {
-        a1 = lua_tostring(L, s);
-        s++;
     }
+    QString name = lua_tostring(L, s);
+    s++;
 
-    QString _name(a1.c_str());
-    std::string luaSendText = "";
+    QString text = "";
     if (n > 1) {
         if (!lua_isstring(L, s)) {
             lua_pushstring(L, "replace: wrong argument type");
             lua_error(L);
             return 1;
-        } else {
-            a2 = lua_tostring(L, s);
         }
+        text = lua_tostring(L, s);
     }
 
     Host& host = getHostFromLua(L);
     if (n == 1) {
-        host.mpConsole->replace(QString(a1.c_str()));
+        host.mpConsole->replace(text);
     } else {
-        mudlet::self()->replace(&host, _name, QString(a2.c_str()));
+        mudlet::self()->replace(&host, name, text);
     }
     return 0;
 }
 
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#deleteLine
 int TLuaInterpreter::deleteLine(lua_State* L)
 {
-    std::string name = "";
+    QString name;
     if (lua_gettop(L) == 1) {
         if (!lua_isstring(L, 1)) {
             lua_pushstring(L, "deleteLine: wrong argument type");
             lua_error(L);
             return 1;
-        } else {
-            name = lua_tostring(L, 1);
         }
+        name = lua_tostring(L, 1);
     }
 
-    QString _name(name.c_str());
     Host& host = getHostFromLua(L);
 
-    if (name.empty()) {
+    if (name.isEmpty() || name == "main") {
```

Comment:
Fair addition, please update documentation!

Reply:
In fact I'm going through this and fix the test everywhere. There's a fair amount of "main" tests both in Lua and in TConsole.cpp that are duplicate and/or inconsistent.

Reply:
As to the Wiki docs, I'll update it when this gets merged.